src/main/scala/mesosphere/marathon/api/Rejection.scala

```diff
@@ -0,0 +1,33 @@
+package mesosphere.marathon
+package api
+
+import javax.servlet.http.HttpServletRequest
+import mesosphere.marathon.plugin.auth.{Authenticator, Authorizer, Identity}
+
+/**
+  * Exception which, when thrown, will cause the corresponding rejection type to be mapped to an API response.
+  *
+  * Mapping is done in [[MarathonExceptionMapper]]
+  */
+case class RejectionException(rejection: Rejection) extends Exception(s"Unhandled rejection: ${rejection}")
+
+/**
+  * Models known reasons why an API request could be confused.
+  */
+sealed trait Rejection {}
+object Rejection {
+  /**
+    * The request was authenticated, but access was not allowed to the resource.
+    */
+  case class AccessDeniedRejection(authorizer: Authorizer, identity: Identity) extends Rejection
```

Comment:
I would personally drop the word Rejection from these classes...

Reply:
The convention was inspired by Akka HTTP, which includes the suffix for each class:

https://doc.akka.io/api/akka-http/current/akka/http/scaladsl/index.html?search=rejection

Also, by convention, Exceptions end with the work "Exception", also.

Does that make it seam okay to you? Or still feel like you would drop it?